[Bug]: compile error at 'global_statistics.c' when using '--disable-dbengine'

### Bug description

I have following compilation error when compiling the v1.44.1 with --disable-dbengine

`configure --disable-unit-tests --disable-cloud --enable-gtests --disable-ebpf --disable-dbengine`

```
  CC       daemon/global_statistics.o
daemon/global_statistics.c: In function ‘global_statistics_charts’:
daemon/global_statistics.c:860:9: error: ‘tier_page_type’ undeclared (first use in this function)
  860 |     if (tier_page_type[0] == PAGE_GORILLA_METRICS)
      |         ^~~~~~~~~~~~~~
daemon/global_statistics.c:860:9: note: each undeclared identifier is reported only once for each function it appears in
daemon/global_statistics.c:860:30: error: ‘PAGE_GORILLA_METRICS’ undeclared (first use in this function)
  860 |     if (tier_page_type[0] == PAGE_GORILLA_METRICS)
      |                              ^~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:8366: daemon/global_statistics.o] Error 1
```


I do not have it when using '--enable-dbengine'
`
configure --disable-unit-tests --disable-cloud --enable-gtests --disable-ebpf --enable-dbengine`

### Expected behavior

having no compilation error :-)

### Steps to reproduce

1. ./configure --disable-unit-tests --disable-cloud --enable-gtests --disable-ebpf --disable-dbengine
2. make
3.
...


### Installation method

from source

### System info

```shell
Linux tst-jan 6.1.0-16-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.67-1 (2023-12-12) x86_64 GNU/Linux
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="12"
/etc/os-release:VERSION="12 (bookworm)"
/etc/os-release:VERSION_CODENAME=bookworm
/etc/os-release:ID=debian
```


### Netdata build info

```shell
na
```


### Additional info

_No response_

--- a/daemon/global_statistics.c
+++ b/daemon/global_statistics.c
@@ -857,6 +857,7 @@ static void global_statistics_charts(void) {
 
     // ----------------------------------------------------------------
 
+#ifdef ENABLE_DBENGINE
     if (tier_page_type[0] == PAGE_GORILLA_METRICS)
     {
         static RRDSET *st_tier0_gorilla_pages = NULL;
@@ -918,6 +919,7 @@ static void global_statistics_charts(void) {
 
         rrdset_done(st_tier0_compression_info);
     }
+#endif
 }
 
 // ----------------------------------------------------------------------------

--- a/daemon/main.c
+++ b/daemon/main.c
@@ -1502,11 +1502,11 @@ int main(int argc, char **argv) {
 #ifdef ENABLE_DBENGINE
                         char* createdataset_string = "createdataset=";
                         char* stresstest_string = "stresstest=";
-#endif
 
                         if(strcmp(optarg, "pgd-tests") == 0) {
                             return pgd_test(argc, argv);
                         }
+#endif
 
                         if(strcmp(optarg, "sqlite-meta-recover") == 0) {
                             sql_init_database(DB_CHECK_RECOVER, 0);

